src/jshint.js

```diff
@@ -4662,9 +4665,8 @@ var JSHINT = (function() {
       // ExportDeclaration :: export Declaration
       this.block = true;
       advance("function");
-      exported[state.tokens.next.value] = ok;
       state.tokens.next.exported = true;
-      state.syntax["function"].fud();
+      state.syntax["function"].fud({ inexport:true });
```

Comment:
I've been comparing with #2078 and this seems a better way of doing it, since the export doesn't have to know about the structure of the function, good.